static/app/views/performance/newTraceDetails/traceWaterfall.tsx

```diff
@@ -590,7 +590,7 @@ export function TraceWaterfall(props: TraceWaterfallProps) {
     if (node) {
       if (isAutogroupedNode(node) && type !== 'ag') {
         if (isParentAutogroupedNode(node)) {
-          node = TraceTree.FindByID(node.head, eventId ?? path!) ?? node;
+          node = TraceTree.FindByID(node.head, path ?? eventId!) ?? node;
         } else if (isSiblingAutogroupedNode(node)) {
           node = node.children.find(n => TraceTree.FindByID(n, eventId ?? path!)) ?? node;
```

Comment:
not sure if this line should also be updated

Reply:
Great catch, I was too focussed on parent autogrouping